Storage/SQL/SQLiteHistory.swift

```diff
@@ -387,24 +394,27 @@ extension SQLiteHistory: BrowserHistory {
             "localVisitDate, remoteVisitDate, localVisitCount, remoteVisitCount,",
             "iconID, iconURL, iconDate, iconType, iconWidth, frecencies",
             "FROM (", query, ")"
-        ].joined(separator: " ")
+            ].joined(separator: " ")
+        return (insertQuery, args)
+    }
 
-        return self.clearTopSitesCache() >>> {
-            return self.db.run(insertQuery, withArgs: args)
-        } >>> {
-            self.prefs.setBool(true, forKey: PrefsKeys.KeyTopSitesCacheIsValid)
-            return succeed()
-        }
+    public func refreshTopsitesQuery(_ limit: Int) -> [(String, Args?)] {
+        return [clearTopsitesQuery(), updateTopSitesCacheQuery(limit)]
     }
 
     public func clearTopSitesCache() -> Success {
```

Comment:
I think `clearTopSitesCache()` can be deleted, or moved to testing code, or removed from the interface, or otherwise removed from the public API.